Production_scripts/B07_Process_District_Nursing_extract.R

```diff
@@ -0,0 +1,196 @@
+
+# Recode HB codes so they match the cost lookup
+dn_extract <- dn_extract %>%
+  mutate(hbtreatcodes = case_when(
+    hbtreatcode == "S08000018" & convert_fyyear_to_year(year) > 2018 ~ "S08000029",
+    hbtreatcode == "S08000027" & convert_fyyear_to_year(year) > 2018 ~ "S08000030",
+    hbtreatcode == "S08000021" & convert_fyyear_to_year(year) > 2019 ~ "S08000031",
+    hbtreatcode == "S08000023" & convert_fyyear_to_year(year) > 2019 ~ "S08000032"
+  )) %>%
+  # sort by hbtreatcode
+  arrange(hbtreatcode)
+
+
+# match files with DN Cost Lookup
+# read in DN cost lookup
+dn_costs_lookup <- haven::read_sav(get_dn_costs_path())
+
+dn_costs_lookup <- dn_costs_lookup %>%
+  select(-hbtreatname) %>%
+  rename(year = "Year") %>%
+  mutate(year = as.numeric(year))
+
+matched_dn_costs <- dn_clean %>%
+  full_join(dn_costs_lookup, by = c("hbtreatcode", "year")) %>%
+  # costs are rough estimates we round them to the nearest pound
+  mutate(cost_total_net = round(cost_total_net, 0))
+
+
+# difference between dates of contacts
+matched_dn_costs <- matched_dn_costs %>%
+  arrange(chi, record_keydate1) %>%
+  group_by(chi) %>%
+  mutate(date_1 = record_keydate1, date_2 = lead(record_keydate1), day_diff = as.numeric(date_2 - date_1)) %>%
+  ungroup() %>%
+  filter(!is.na(day_diff))
+
+
+# continuous care marker
+matched_dn_costs <- matched_dn_costs %>%
+  tibble::add_column(ccm = 1) %>%
+  group_by(chi) %>%
+  mutate(ccm = if_else(day_diff < 7 | day_diff == 7, lead(ccm), lead(ccm) + 1)) %>%
+  ungroup()
+
+
+# costs per month
+matched_dn_costs <- matched_dn_costs %>%
+  create_day_episode_costs(record_keydate1, cost_total_net)
+
+
+## save outfile ---------------------------------------
+
+outfile <- matched_dn_costs %>%
+  group_by(year, chi, recid, smr_type, ccm) %>%
+  summarise(
+    record_keydate1 = min(record_keydate1),
+    record_keydate2 = max(record_keydate2),
+    dob = last(dob),
+    hbtreatcode = last(hbtreatcode),
+    hbrescode = last(hbrescode),
+    hscp = last(hscp),
+    lca = last(lca),
+    datazone = last(datazone),
+    age = last(age),
+    diag1 = first(primary_intervention),
+    diag2 = first(intervention_1),
+    diag3 = first(intervention_2),
+    diag4 = last(primary_intervention),
+    diag5 = last(intervention_1),
+    diag6 = last(intervention_2),
+    postcode = last(postcode),
+    gender = first(gender),
+    gpprac = first(gpprac),
+    cost_total_net = sum(cost_total_net),
+    location = first(location_contact),
+    jan_cost = sum(jan_cost),
+    feb_cost = sum(feb_cost),
+    mar_cost = sum(mar_cost),
+    apr_cost = sum(apr_cost),
+    may_cost = sum(may_cost),
+    jun_cost = sum(jun_cost),
+    jul_cost = sum(jul_cost),
+    aug_cost = sum(aug_cost),
+    sep_cost = sum(sep_cost),
+    oct_cost = sum(oct_cost),
+    nov_cost = sum(nov_cost),
+    dec_cost = sum(dec_cost)
+  ) %>%
+  ungroup() %>%
+  # factor
+  mutate(across(c(location, starts_with("diag")),
+    .x = as.factor(.x)
+  ))
```

Comment:
If this is a factor, then I think we should specify all the possible levels, as it's quite likely that some years won't contain all possible values. Jen or I could get an extract of all possible values from BOXI which can be used for this.

Reply:
@Moohan do we need to define as a factor?

Reply:
Good point, probably don't define this as a factor as it will have 100's if not thousands of levels.